### DISCUSSION TITLE
Understanding the Passcode, Request for more control of timeouts

### DISCUSSION THREAD

**Discussion by arcanecode** [standardnotes/forum#2542] [Suggestions]
# Introduction

I guess I'm slow to the game, as I just discoverd the Passcode Lock. What a great feature! It solves a concern of mine, namely what if for some reason I no longer have access to a computer with my Standard Notes installation. Maybe the computer crashes, or I'm suddenly laid off. 

With a passcode I can set a timer so that without the passcode you can't get in, and while passcodes aren't quite as good as a super secure gibberish password, they are easier to use and thus will be more likely to be used. Now if my computer dies, I can feel safer knowing the repair person won't be able to boot my machine and get into my notes. 

So I have one question, and one suggestion.

# Question

First, it seems like the Passcode is unique to the device I implement it on. For example, if I create a passcode on my laptop, the passcode is not carried over to my iPad. 

I actually like this as it lets me set a passcode on my iPad of 5 minutes, my laptop of 1 hour, and on my desktop that never leaves the house not have a passcode at all. 

I always want to make sure I understand the security features of any application I use, so wanted to confirm I have it right, and that passcodes are unique to the device it is created on.

# Suggestion

So for the suggestion, it might be nice to have a few other options for a time out. 5 minutes seems a bit brief, but 1 hour may be a bit too long. Maybe a 15 and 30 minute option as well. I could also see setting a 4 hour time, to get me though my morning at work, 8 hours for all day, or just 17 minutes because I'm the paranoid type.

Or even better, just have an option to just let the user key in the length of time. Have a text box for them to enter a number, then radio option buttons for the user to indicate if that number is minutes or hours. 

If a user enters 0, you can either not allow it, or inform them a value of 0 will disable the passcode, is that what they wanted.

# Other

By the way, the laid off thing is a valid concern for a lot of us consultants. I was on a contract about 15 years ago for a medium size company when one day an email pops up. It said the company had just been aquired as part of a hostile takeover, and that effective immediately all consulting contracts were on hold. We had one hour to clear the building, and they'd be in contact with our respective firms. Fortunately this gave me time to remove anything personal I had on there, which wasn't much but we all have our own favorite tools we like to use. 

A co-worker (from the same consulting firm) said it was even worse for him. I was at the annex on the edge of town, but he was in the main headquarters. He told me he just got a tap on the shoulder, was informed he was now a goner and he was not allowed to touch his computer. He could only grab his personal stuff (books etc) and was walked to the door.

**Comment by atmoio** [comment]
Thanks for the detailed suggestion. More fine-grained lockout options have indeed been suggested before and could be useful. We'll have to explore around this area more in the future. I don't think there are any immediate plans to do work in this area, but the next major iteration of passcode will definitely want to consider the points you brought up here.

**Comment by tiddesk** [comment]
I think the option to let the user key in the length of time is really the way to go, although I'd also ask for the ability to set the lock to engage regardless of timer status when the desktop is locked (manually or by idle timeout)
